### PR TITLE
improve tests of api/components

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java [hunk 1]
```diff
@@ -84,4 +85,91 @@ public void search_projects_without_sort() {
...
   }
 
+  @Test
+  public void show() {
+    String key = randomAlphanumeric(20);
+    String id = randomAlphanumeric(20);
+    underTest.show(new ShowWsRequest()
+      .setKey(key)
+      .setId(id));
+
```
Comment (julienlancelot) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Please add checks on parser :
`assertThat(serviceTester.getGetParser()).isSameAs(ShowWsResponse);`

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java [hunk 2]
```diff
@@ -84,4 +85,91 @@ public void search_projects_without_sort() {
...
+  @Test
+  public void suggestions() {
+    SuggestionsWsRequest.More more = SuggestionsWsRequest.More.BRC;
+    String s = randomAlphanumeric(20);
+    underTest.suggestions(SuggestionsWsRequest.builder()
+      .setMore(more)
+      .setS(s)
+      .setRecentlyBrowsed(asList("key-1", "key-2"))
+      .build());
+
```
Comment (julienlancelot) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Please add check on parser

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java [hunk 3]
```diff
@@ -84,4 +85,91 @@ public void search_projects_without_sort() {
...
+    int pageSize = 39;
+    String textQuery = randomAlphanumeric(20);
+    underTest.search(new SearchWsRequest()
+      .setOrganization(organization)
+      .setQualifiers(asList("q1", "q2"))
+      .setPage(page)
+      .setPageSize(pageSize)
+      .setQuery(textQuery)
+    );
+
```
Comment (julienlancelot) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Please add check on parser

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java [hunk 4]
```diff
@@ -84,4 +85,91 @@ public void search_projects_without_sort() {
...
+    underTest.tree(new TreeWsRequest()
+      .setBaseComponentId(componentId)
+      .setBaseComponentKey(componentKey)
+      .setQualifiers(asList("q1", "q2"))
+      .setStrategy(strategy)
+      .setPage(page)
+      .setPageSize(pageSize)
+      .setQuery(query)
+      .setSort(asList("sort1", "sort2")));
+
```
Comment (julienlancelot) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Please add check on parser

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java (hunk 1 above)
Comment (bartfastiel) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Done.

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java (hunk 2 above)
Comment (bartfastiel) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Done.

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java (hunk 3 above)
Comment (bartfastiel) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Done.

File: sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java (hunk 4 above)
Comment (bartfastiel) [inline sonar-ws/src/test/java/org/sonarqube/ws/client/component/ComponentsServiceTest.java]: Done.

**Pull request merged**